hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/block/TestDeletedBlockLog.java

```diff
@@ -527,6 +533,22 @@ public void testDeletedBlockTransactions()
     Assertions.assertEquals(2, blocks.size());
   }
 
+  @Test
+  public void testDeletedBlockTransactionsOfDeletedContainer()
+      throws IOException, TimeoutException {
+    int txNum = 10;
+    List<DeletedBlocksTransaction> blocks;
+
+    // Creates {TXNum} TX in the log.
+    Map<Long, List<Long>> deletedBlocks = generateData(txNum,
```

Comment:
Containers are deleted only if block count is zero. In my opinion you could also have an integration testcase with underreplicated containers & going ahead deleting these blocks. This particular issue can happen then.

Reply:
@swamirishi Thank you for the analysis, I added some comments in the PR instead of adding an integrated test. I tried to add the test earlier, but then I found the test is too time-costly, if we have too many tests for corner cases, the whole CI would become huge and long, so maybe a comment is enough here. What do you think?

Reply:
I think a comment could be added in the commitTransactions function instead where the record is being removed.

Reply:
Sorry for late reply, but I don't see the relationship between `commitTransactions` and the cleaning of deleted containers.

Reply:
Sorry for holding this PR for pretty long. The comment here could be anywhere I suggested in the commitTransaction function as there is already a comment around the same but not covering under replication part. The comment in this class should also be fine.